pkg/apis/v1/nodeclaim.go

```diff
@@ -146,14 +146,14 @@ type KubeletConfiguration struct {
 
 type NodeClassReference struct {
 	// Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
-	// +optional
-	Kind string `json:"kind,omitempty"`
+	// +required
```

Comment:
Should we also require that these strings have a minimum length so that they also can't be an empty string?

Reply:
Can we leave that to CodeClass readiness on the NodePool to validate if the nodeClassRef is valid?

Reply:
Sure, I'm more just thinking that required fields shouldn't be empty anyways so it may just be better to validate that they aren't empty earlier